sematic/plugins/building/docker_client_utils.py

```diff
@@ -0,0 +1,72 @@
+"""
+Utility code for interacting with the `docker-py` client.
+"""
+# Standard Library
+import sys
+import textwrap
+from typing import Any, Dict, Generator, List, Optional, TextIO
+
+_ROLLING_PRINT_COLUMNS = 120
+_ROLLING_PRINT_ROWS = 10
+
+
+def _rolling_print_status_updates(
+    status_updates: Generator[Dict[str, Any], None, None],
+    output_handle: TextIO = sys.stderr,
+    columns: int = _ROLLING_PRINT_COLUMNS,
+    rows: int = _ROLLING_PRINT_ROWS,
+) -> Optional[Dict[str, Any]]:
+    """
+    Prints a rolling window of Docker server status message updates.
+    """
+    message_window: List[str] = []
+
+    for status_update in status_updates:
+
+        if "error" in status_update.keys():
+            return status_update
+
+        update_str = _docker_status_update_to_str(status_update)
+        if update_str is None:
+            continue
+
+        line_count = len(message_window)
+        for raw_line in update_str.split("\n"):
+            wrapped_lines = textwrap.wrap(
+                raw_line, replace_whitespace=False, width=columns
+            )
+            message_window += wrapped_lines
```

Comment:
should `line_count` be updated after this?

Reply:
Renamed `line_count` to `previous_line_count` to clarify.